qgis_hub_plugin/gui/resource_browser.py

```diff
@@ -174,30 +179,40 @@ def show_preview(self):
 
     def download_resource(self):
         resource = self.selected_resource()
-        file_path = download_resource_file(
-            resource.file, resource.resource_type, resource.uuid
-        )
+        file_extension = os.path.splitext(resource.file)[1]
+
+        file_path = QFileDialog.getSaveFileName(
+            self,
+            self.tr("Save Resource"),
+            resource.file,
+            self.tr(
+                "All Files (*);;Geopackage (*.gpkg);;QGIS Model (*.model3);; ZIP Files (*.zip)"
+            ),
+        )[0]
+
         if file_path:
-            text = f"Successfully download {resource.name} to {file_path}"
+            if not file_path.endswith(file_extension):
+                file_path = file_path + file_extension
+
+            download_resource_file(resource.file, file_path)
+
+            text = self.tr(f"Successfully downloaded {resource.name} to {file_path}")
+
             self.iface.messageBar().pushMessage(self.tr("Success"), text, duration=5)
             if self.checkBoxOpenDirectory.isChecked():
-                QDesktopServices.openUrl(QUrl.fromLocalFile(str(file_path.parent)))
+                QDesktopServices.openUrl(
+                    QUrl.fromLocalFile(str(Path(file_path).parent))
+                )
         else:
-            text = f"Failed download {resource.name} from {resource.file}"
+            text = self.tr(f"Download canceled for {resource.name}")
```

Comment:
Use string format, not f-string, because it's not translatable easily/correctly

Reply:
Actually, no need to tell the user that the user cancels the download, it's too much for the user. It's better to remove it.

Previously, the message is about failure to download (that's important), but canceling to download is not needed. See my other comment for putting failure to download instead.

Reply:
> Use string format, not f-string, because it's not translatable easily/correctly

Automatically changing to f-string during commit.

Reply:
ok, let's leave it for now. Perhaps they already support it. I will check the translation system.